packages/core/src/controllers/relayer.ts

```diff
@@ -168,15 +181,14 @@ export class Relayer extends IRelayer {
       await Promise.all([
         new Promise<void>((resolve) => {
           if (!this.initialized) resolve();
-
           // wait for the subscriber to finish resubscribing to its topics
           this.subscriber.once(SUBSCRIBER_EVENTS.resubscribed, () => {
             resolve();
           });
         }),
         await Promise.race([
           new Promise<void>(async (resolve) => {
-            await this.provider.connect();
+            await createExpiringPromise(this.provider.connect(), 5_000, "socket hang up");
```

Comment:
Are we shadowing the native/built-in `socket hang up` error with this and muddying the waters between whether the relayer triggered this or whether it came from the socket itself?

Might be less confusing for us to use a differentiated string here like `Provider failed to connect`. Lmk if I'm missing something here.

Reply:
Guessing we need to expand on https://github.com/WalletConnect/walletconnect-monorepo/pull/2108/files#diff-2666bbe86db232da1fe48efc66faddc72893a26741e59bcc7510ea3818bbd581R205 if we want this to throw the way `socket hang up` does but with a different message/cause.